web/docs/guides/websockets.md

```diff
@@ -0,0 +1,129 @@
+---
+title: WebSockets
+---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# WebSocket support
+Wasp provides a fully integrated WebSocket experience by utilizing [Socket.IO](https://socket.io/) on the client and server. We handle making sure your URLs are correctly setup, CORS is enabled, and provide a useful `useSocket` abstraction for use in React components.
+
+## Enable in Wasp file
+Here, we specify we are using WebSockets by adding `webSocket` to our `app` and providing the required `fn`. You can optionally specify a list of `entities` for use in your context (like in operations).
+
+```wasp title=todoApp.wasp
+app todoApp {
+  // ...
+
+  webSocket: {
+    fn: import { webSocketFn } from "@server/webSocket.js"
+    // Optional: entities: [...]
+  },,
+}
+```
+
+## Optional: Add shared types for client and server use
+If TypeScript is your thing, you can add shared interface definitions to use on both the client and server. This allows you to type all the messages you send and receive!
+
+```ts title=src/shared/webSocket.ts
+import { WaspSocketData } from '@wasp/universal/types'
+
+export interface ServerToClientEvents {
+  chatMessage: (msg: { id: string, username: string, text: string }) => void;
+}
+
+export interface ClientToServerEvents {
+  chatMessage: (msg: string) => void;
+}
+
+export interface InterServerEvents { }
+
+// Data that is attached to the socket.
+// NOTE: Wasp automatically injects the JWT into the connection,
+// and if present/valid, the server adds a user to the socket.
+export interface SocketData extends WaspSocketData { }
+```
+
+## Add server support
+On the server, you will get `io` argument and `context` for your WebSocket function. You can use this `io` object to register callbacks for all the regular Socket.IO events.
```

Comment:
Will they know what is this `io` object? Is it something they would normally get from socket.io? Could we link to what this object is, maybe link to its type on socket.io or its docs or something?

Reply:
You don't say much about `context`?

Reply:
I think they should, but I can link to Socket.IO docs for `io`. I can mention `context` will contain entities. And yeah, I can note the user on socket 👍🏻